drivers/sx126x/sx126x.c

```diff
@@ -427,17 +428,94 @@ int sx126x_init(sx126x_t *dev)
     return res;
 }
 
+sx126x_chip_modes_t sx126x_get_state(const sx126x_t *dev)
+{
+    sx126x_chip_status_t radio_status;
+    sx126x_get_status(dev, &radio_status);
+    return radio_status.chip_mode;
+}
+
+void sx126x_set_state(sx126x_t *dev, sx126x_chip_modes_t state)
+{
+    switch (state) {
+    case SX126X_CHIP_MODE_TX:
+        sx126x_set_tx(dev, 0); /* no TX frame timeout */
+        break;
+    case SX126X_CHIP_MODE_RX:
+        if (dev->rx_timeout >= 0) {
+            int timeout = (sx126x_symbol_to_msec(dev, dev->rx_timeout));
+            sx126x_set_rx_tx_fallback_mode(dev, SX126X_FALLBACK_STDBY_XOSC);
+            if (timeout > 0) {
+                sx126x_set_rx(dev, timeout);
+            }
+            else {
+                sx126x_set_rx(dev, SX126X_RX_SINGLE_MODE);
+            }
+        }
+        else {
+            /* By default, the radio will always return in STDBY_RC
+               unless the configuration is changed by using this command.
+               Changing the default mode from STDBY_RC to STDBY_XOSC or FS
+               will only have an impact on the switching time of the radio */
+            sx126x_set_rx_tx_fallback_mode(dev, SX126X_FALLBACK_FS);
+            sx126x_set_rx_with_timeout_in_rtc_step(dev, SX126X_RX_CONTINUOUS);
+        }
+        break;
+    case SX126X_CHIP_MODE_STBY_RC:
+    case SX126X_CHIP_MODE_STBY_XOSC:
+        sx126x_set_standby(dev, state);
+        break;
+    default:
+        break;
+    }
+}
+
 uint32_t sx126x_get_channel(const sx126x_t *dev)
 {
     DEBUG("[sx126x]: sx126x_get_radio_status \n");
     return dev->channel;
 }
 
+/* 9.2.1 Image Calibration for Specific Frequency Bands */
+static void _cal_img(sx126x_t *dev, uint32_t freq)
+{
+    sx126x_chip_modes_t state = sx126x_get_state(dev);
+    /* don't know what to do with frequencies that don't fit in the intervals from the datasheet */
+    if (freq <= MHZ(928) && freq >= MHZ(902)) {
```

Comment:
I know it's semantically the same, but writing it like 
```suggestion
    if (freq >= MHZ(902) && freq <= MHZ(928)) {
```
avoids a contortion in my head.